core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -247,14 +255,63 @@ private TrustAnchor getTrustAnchorForPublicKey(final JsonObject keyProps) {
                 final X509EncodedKeySpec keySpec = new X509EncodedKeySpec(encodedKey);
                 final KeyFactory factory = KeyFactory.getInstance(type);
                 final PublicKey publicKey = factory.generatePublic(keySpec);
-                return new TrustAnchor(subjectDn, publicKey, null);
+                final RevocableTrustAnchor trustAnchor = new RevocableTrustAnchor(subjectDn, publicKey, null);
+                if (isRevocable(keyProps)) {
+                    setRevocationProperties(trustAnchor, keyProps);
+                }
+                return trustAnchor;
             } catch (final GeneralSecurityException e) {
                 LOG.debug("failed to instantiate trust anchor's public key", e);
                 return null;
             }
         }
     }
 
+    private TrustAnchor getTrustAnchorForCertificate(final JsonObject keyProps) {
+        final X509Certificate certificate;
+        final CertificateFactory certificateFactory;
+        final byte[] encodedCertificate = getProperty(keyProps, TenantConstants.FIELD_PAYLOAD_CERT, byte[].class);
+        try {
+            certificateFactory = CertificateFactory.getInstance("X.509");
+            certificate = (X509Certificate) certificateFactory.generateCertificate(
+                    new ByteArrayInputStream(encodedCertificate));
+            final RevocableTrustAnchor trustAnchor = new RevocableTrustAnchor(certificate, null);
+            if (isRevocable(keyProps)) {
+                setRevocationProperties(trustAnchor, keyProps);
+            }
+            return trustAnchor;
+        } catch (CertificateException e) {
+            LOG.error("failed to parse Trusted CA", e);
+            return null;
+        }
+    }
+
+    private boolean isRevocable(final JsonObject keyProps) {
+        return getProperty(keyProps, TenantConstants.FIELD_OCSP_REVOCATION_ENABLED, Boolean.class, false);
+    }
+
+    private void setRevocationProperties(final RevocableTrustAnchor trustAnchor, final JsonObject keyProps) {
```

Comment:
now that we need a custom TrustAnchor class, we might as well move all this code for configuring it to the RevocableTrustAnchor class itself, right? For example by means of adding a constrictor which simply accepts a JsonObject and creates the configured RevocableTrustAnchor ...